### ISSUE TITLE
Better handling of numpy.int with custom forces

### ISSUE THREAD

**Issue by craldaz** [openmm/openmm#2730]
I am trying to add a torsion restraint to a list of torsions. 

```python3
    tforce = openmm.CustomTorsionForce("0.5*k*(1-cos(theta-theta0))")
    tforce.addPerTorsionParameter("k")
    tforce.addPerTorsionParameter("theta0")
    system.addForce(tforce)
    torsions=np.loadtxt('restrain_torsions.txt',dtype=int)
    for t in torsions:
        d = Dihedral(t[0],t[1],t[2],t[3])    # Not an openmm function
        theta_0 = d.value(xyz)
        tforce.addTorsion(t[0],t[1],t[2],t[3],[theta_0,100.])      # <-- ERROR HERE
```

However, I receive this error which seems to suggest my input arguments are wrong or otherwise the function isn't extended to python yet 

```python3
    tforce.addTorsion(t[0],t[1],t[2],t[3],[theta_0,100.])
  File "/home/caldaz/.conda/envs/py3/lib/python3.6/site-packages/simtk/openmm/openmm.py", line 13491, in addTorsion
    return _openmm.CustomTorsionForce_addTorsion(self, *args)
NotImplementedError: Wrong number or type of arguments for overloaded function 'CustomTorsionForce_addTorsion'.
  Possible C/C++ prototypes are:
    OpenMM::CustomTorsionForce::addTorsion(int,int,int,int,std::vector< double,std::allocator< double > > const &)
    OpenMM::CustomTorsionForce::addTorsion(int,int,int,int)
```

In this[ tutorial](http://www.maccallumlab.org/news/2015/1/23/testing)  they perform something similar except with a CustomBondForce

**Comment by craldaz** [comment]
It was because I used a numpy integer 64.

**Comment by jchodera** [comment]
I'm reopening this since we've been burned by this as well. We should be able to make the Python layer more tolerant of numpy `int` types, casting them to Python `int`s as needed. 

I've relabeled the issue to reflect this.

**Comment by tristanic** [comment]
> we've been burned by this as well. 

*raises hand* Yup. Me too.

**Comment by peastman** [comment]
This is fixed by #2961.
